diff -ruN riscv/riscv64-unknown-elf/include/c++/14.2.0/bits/std_abs.h riscv-modified/riscv64-unknown-elf/include/c++/14.2.0/bits/std_abs.h
--- riscv/riscv64-unknown-elf/include/c++/14.2.0/bits/std_abs.h	2025-03-10 21:27:45.503332094 -0700
+++ riscv-modified/riscv64-unknown-elf/include/c++/14.2.0/bits/std_abs.h	2025-03-17 15:03:30.948683661 -0700
@@ -35,9 +35,9 @@
 #include <bits/c++config.h>
 
 #define _GLIBCXX_INCLUDE_NEXT_C_HEADERS
-#include <stdlib.h>
+#include_next <stdlib.h>
 #ifdef __CORRECT_ISO_CPP_MATH_H_PROTO
-# include <math.h>
+# include_next <math.h>
 #endif
 #undef _GLIBCXX_INCLUDE_NEXT_C_HEADERS
 
diff -ruN riscv/riscv64-unknown-elf/include/c++/14.2.0/cmath riscv-modified/riscv64-unknown-elf/include/c++/14.2.0/cmath
--- riscv/riscv64-unknown-elf/include/c++/14.2.0/cmath	2025-03-10 21:31:01.626970093 -0700
+++ riscv-modified/riscv64-unknown-elf/include/c++/14.2.0/cmath	2025-03-17 15:03:31.244688946 -0700
@@ -44,7 +44,7 @@
 #include <bits/cpp_type_traits.h>
 #include <ext/type_traits.h>
 #define _GLIBCXX_INCLUDE_NEXT_C_HEADERS
-#include <math.h>
+#include_next <math.h>
 #undef _GLIBCXX_INCLUDE_NEXT_C_HEADERS
 #include <bits/std_abs.h>
 
@@ -2089,149 +2089,145 @@
   // functions
   using ::acosh;
   using ::acoshf;
+  using ::acoshl;
 
   using ::asinh;
   using ::asinhf;
+  using ::asinhl;
 
   using ::atanh;
   using ::atanhf;
+  using ::atanhl;
 
   using ::cbrt;
   using ::cbrtf;
+  using ::cbrtl;
 
   using ::copysign;
   using ::copysignf;
+  using ::copysignl;
 
   using ::erf;
   using ::erff;
+  using ::erfl;
 
   using ::erfc;
   using ::erfcf;
+  using ::erfcl;
 
   using ::exp2;
   using ::exp2f;
+  using ::exp2l;
 
   using ::expm1;
   using ::expm1f;
+  using ::expm1l;
 
   using ::fdim;
   using ::fdimf;
+  using ::fdiml;
 
   using ::fma;
   using ::fmaf;
+  using ::fmal;
 
   using ::fmax;
   using ::fmaxf;
+  using ::fmaxl;
 
   using ::fmin;
   using ::fminf;
+  using ::fminl;
 
   using ::hypot;
   using ::hypotf;
+  using ::hypotl;
 
   using ::ilogb;
   using ::ilogbf;
+  using ::ilogbl;
 
   using ::lgamma;
   using ::lgammaf;
+  using ::lgammal;
 
 #ifndef _GLIBCXX_NO_C99_ROUNDING_FUNCS
   using ::llrint;
   using ::llrintf;
+  using ::llrintl;
 
   using ::llround;
   using ::llroundf;
+  using ::llroundl;
 #endif
 
   using ::log1p;
   using ::log1pf;
+  using ::log1pl;
 
   using ::log2;
   using ::log2f;
+  using ::log2l;
 
   using ::logb;
   using ::logbf;
+  using ::logbl;
 
   using ::lrint;
   using ::lrintf;
+  using ::lrintl;
 
   using ::lround;
   using ::lroundf;
+  using ::lroundl;
 
   using ::nan;
   using ::nanf;
+  using ::nanl;
 
   using ::nearbyint;
   using ::nearbyintf;
+  using ::nearbyintl;
 
   using ::nextafter;
   using ::nextafterf;
+  using ::nextafterl;
+
+  using ::nexttoward;
+  using ::nexttowardf;
+  using ::nexttowardl;
 
   using ::remainder;
   using ::remainderf;
+  using ::remainderl;
 
   using ::remquo;
   using ::remquof;
+  using ::remquol;
 
   using ::rint;
   using ::rintf;
+  using ::rintl;
 
   using ::round;
   using ::roundf;
+  using ::roundl;
 
   using ::scalbln;
   using ::scalblnf;
+  using ::scalblnl;
 
   using ::scalbn;
   using ::scalbnf;
+  using ::scalbnl;
 
   using ::tgamma;
   using ::tgammaf;
+  using ::tgammal;
 
   using ::trunc;
   using ::truncf;
-
-#ifdef _GLIBCXX_USE_C99_MATH_LONG_DOUBLE_FUNCS
-  using ::acoshl;
-  using ::asinhl;
-  using ::atanhl;
-  using ::cbrtl;
-  using ::copysignl;
-  using ::erfl;
-  using ::erfcl;
-  using ::exp2l;
-  using ::expm1l;
-  using ::fdiml;
-  using ::fmal;
-  using ::fmaxl;
-  using ::fminl;
-  using ::hypotl;
-  using ::ilogbl;
-  using ::lgammal;
-#ifndef _GLIBCXX_NO_C99_ROUNDING_FUNCS
-  using ::llrintl;
-  using ::llroundl;
-#endif
-  using ::log1pl;
-  using ::log2l;
-  using ::logbl;
-  using ::lrintl;
-  using ::lroundl;
-  using ::nanl;
-  using ::nearbyintl;
-  using ::nextafterl;
-  using ::nexttoward;
-  using ::nexttowardf;
-  using ::nexttowardl;
-  using ::remainderl;
-  using ::remquol;
-  using ::rintl;
-  using ::roundl;
-  using ::scalblnl;
-  using ::scalbnl;
-  using ::tgammal;
   using ::truncl;
-#endif
 
   /// Additional overloads.
 #ifndef __CORRECT_ISO_CPP11_MATH_H_PROTO_FP
diff -ruN riscv/riscv64-unknown-elf/include/c++/14.2.0/cstdlib riscv-modified/riscv64-unknown-elf/include/c++/14.2.0/cstdlib
--- riscv/riscv64-unknown-elf/include/c++/14.2.0/cstdlib	2025-03-10 21:31:36.979626074 -0700
+++ riscv-modified/riscv64-unknown-elf/include/c++/14.2.0/cstdlib	2025-03-17 15:03:31.232688733 -0700
@@ -76,7 +76,7 @@
 // Need to ensure this finds the C library's <stdlib.h> not a libstdc++
 // wrapper that might already be installed later in the include search path.
 #define _GLIBCXX_INCLUDE_NEXT_C_HEADERS
-#include <stdlib.h>
+#include_next <stdlib.h>
 #undef _GLIBCXX_INCLUDE_NEXT_C_HEADERS
 #include <bits/std_abs.h>
 
diff -ruN riscv/riscv64-unknown-elf/include/c++/14.2.0/riscv64-unknown-elf/bits/c++config.h riscv-modified/riscv64-unknown-elf/include/c++/14.2.0/riscv64-unknown-elf/bits/c++config.h
--- riscv/riscv64-unknown-elf/include/c++/14.2.0/riscv64-unknown-elf/bits/c++config.h	2025-03-15 12:06:27.309124059 -0700
+++ riscv-modified/riscv64-unknown-elf/include/c++/14.2.0/riscv64-unknown-elf/bits/c++config.h	2025-03-17 15:03:31.208688302 -0700
@@ -1555,7 +1555,7 @@
 /* Define if compatibility should be provided for -mlong-double-64. */
 
 /* Define to the letter to which size_t is mangled. */
-#define _GLIBCXX_MANGLE_SIZE_T j
+#define _GLIBCXX_MANGLE_SIZE_T m
 
 /* Define if C99 llrint and llround functions are missing from <math.h>. */
 /* #undef _GLIBCXX_NO_C99_ROUNDING_FUNCS */
@@ -1564,13 +1564,13 @@
 /* #undef _GLIBCXX_NO_SLEEP */
 
 /* Define if ptrdiff_t is int. */
-#define _GLIBCXX_PTRDIFF_T_IS_INT 1
+/* #undef _GLIBCXX_PTRDIFF_T_IS_INT */
 
 /* Define if using setrlimit to set resource limits during "make check" */
 /* #undef _GLIBCXX_RES_LIMITS */
 
 /* Define if size_t is unsigned int. */
-#define _GLIBCXX_SIZE_T_IS_UINT 1
+/* #undef _GLIBCXX_SIZE_T_IS_UINT */
 
 /* Define if static tzdata should be compiled into the library. */
 #define _GLIBCXX_STATIC_TZDATA 1
diff -ruN riscv/riscv64-unknown-elf/include/c++/14.2.0/tr1/cmath riscv-modified/riscv64-unknown-elf/include/c++/14.2.0/tr1/cmath
--- riscv/riscv64-unknown-elf/include/c++/14.2.0/tr1/cmath	2025-03-10 21:33:35.713850706 -0700
+++ riscv-modified/riscv64-unknown-elf/include/c++/14.2.0/tr1/cmath	2025-03-17 15:03:32.396709519 -0700
@@ -162,145 +162,143 @@
   // functions
   using ::acosh;
   using ::acoshf;
+  using ::acoshl;
 
   using ::asinh;
   using ::asinhf;
+  using ::asinhl;
 
   using ::atanh;
   using ::atanhf;
+  using ::atanhl;
 
   using ::cbrt;
   using ::cbrtf;
+  using ::cbrtl;
 
   using ::copysign;
   using ::copysignf;
+  using ::copysignl;
 
   using ::erf;
   using ::erff;
+  using ::erfl;
 
   using ::erfc;
   using ::erfcf;
+  using ::erfcl;
 
   using ::exp2;
   using ::exp2f;
+  using ::exp2l;
 
   using ::expm1;
   using ::expm1f;
+  using ::expm1l;
 
   using ::fdim;
   using ::fdimf;
+  using ::fdiml;
 
   using ::fma;
   using ::fmaf;
+  using ::fmal;
 
   using ::fmax;
   using ::fmaxf;
+  using ::fmaxl;
 
   using ::fmin;
   using ::fminf;
+  using ::fminl;
 
   using ::hypot;
   using ::hypotf;
+  using ::hypotl;
 
   using ::ilogb;
   using ::ilogbf;
+  using ::ilogbl;
 
   using ::lgamma;
   using ::lgammaf;
+  using ::lgammal;
 
   using ::llrint;
   using ::llrintf;
+  using ::llrintl;
 
   using ::llround;
   using ::llroundf;
+  using ::llroundl;
 
   using ::log1p;
   using ::log1pf;
+  using ::log1pl;
 
   using ::log2;
   using ::log2f;
+  using ::log2l;
 
   using ::logb;
   using ::logbf;
+  using ::logbl;
 
   using ::lrint;
   using ::lrintf;
+  using ::lrintl;
 
   using ::lround;
   using ::lroundf;
+  using ::lroundl;
 
   using ::nan;
   using ::nanf;
+  using ::nanl;
 
   using ::nearbyint;
   using ::nearbyintf;
+  using ::nearbyintl;
 
   using ::nextafter;
   using ::nextafterf;
+  using ::nextafterl;
+
+  using ::nexttoward;
+  using ::nexttowardf;
+  using ::nexttowardl;
 
   using ::remainder;
   using ::remainderf;
+  using ::remainderl;
 
   using ::remquo;
   using ::remquof;
+  using ::remquol;
 
   using ::rint;
   using ::rintf;
+  using ::rintl;
 
   using ::round;
   using ::roundf;
+  using ::roundl;
 
   using ::scalbln;
   using ::scalblnf;
+  using ::scalblnl;
 
   using ::scalbn;
   using ::scalbnf;
+  using ::scalbnl;
 
   using ::tgamma;
   using ::tgammaf;
+  using ::tgammal;
 
   using ::trunc;
   using ::truncf;
-
-#ifdef _GLIBCXX_USE_C99_MATH_LONG_DOUBLE_FUNCS
-  using ::acoshl;
-  using ::asinhl;
-  using ::atanhl;
-  using ::cbrtl;
-  using ::copysignl;
-  using ::erfl;
-  using ::erfcl;
-  using ::exp2l;
-  using ::expm1l;
-  using ::fdiml;
-  using ::fmal;
-  using ::fmaxl;
-  using ::fminl;
-  using ::hypotl;
-  using ::ilogbl;
-  using ::lgammal;
-  using ::llrintl;
-  using ::llroundl;
-  using ::log1pl;
-  using ::log2l;
-  using ::logbl;
-  using ::lrintl;
-  using ::lroundl;
-  using ::nanl;
-  using ::nearbyintl;
-  using ::nextafterl;
-  using ::nexttoward;
-  using ::nexttowardf;
-  using ::nexttowardl;
-  using ::remainderl;
-  using ::remquol;
-  using ::rintl;
-  using ::roundl;
-  using ::scalblnl;
-  using ::scalbnl;
-  using ::tgammal;
   using ::truncl;
-#endif
 
 #endif
 
